### PR TITLE
Add test 'CRUD order status'

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Add test 'CRUD order status'
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | TEST_PATH="functional/BO/13_shopParameters/02_orderSettings/02_statuses/01_CRUDOrderStatus" URL_FO=yourShopURL npm run specific-test

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/21459)

### CODE REVIEW

File: tests/UI/pages/BO/orders/view.js [hunk 1]
```diff
@@ -83,6 +83,26 @@ class Order extends BOBasePage {
     return this.getTextContent(page, `${this.orderStatusesSelect} option[selected='selected']`, false);
   }
 
+  /**
+   * Is status exist
+   * @param page
+   * @param statusName
+   * @returns {Promise<boolean>}
+   */
+  async isStatusExist(page, statusName) {
```
Comment (SimonGrn) [inline tests/UI/pages/BO/orders/view.js]: I'm not a perfect english speaker but something feels off with this name. @matks do you have a suggestion ?

File: tests/UI/pages/BO/shopParameters/orderSettings/index.js [hunk 2]
```diff
@@ -26,7 +29,7 @@ class OrderSettings extends BOBasePage {
     this.giftWrappingPriceInput = '#form_gift_options_gift_wrapping_price';
     this.giftWrappingTaxSelect = '#form_gift_options_gift_wrapping_tax_rules_group';
     this.recycledPackagingToggle = toggle => `${this.generalForm}`
-    + ` label[for='form_gift_options_offer_recyclable_pack_${toggle}']`;
+      + ` label[for='form_gift_options_offer_recyclable_pack_${toggle}']`;
```
Comment (SimonGrn) [inline tests/UI/pages/BO/shopParameters/orderSettings/index.js:32]: linter suggested that ?

File: tests/UI/pages/BO/shopParameters/orderSettings/statuses/add.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    this.logableOnCheckbox = '#logable_on';
+    this.invoiceOnCheckbox = '#invoice_on';
+    this.hiddenOnCheckbox = '#hidden_on';
+    this.sendEmailOnCheckbox = '#send_email_on';
+    this.pdfInvoiceOnCheckbox = '#pdf_invoice_on';
+    this.pdfDeliveryOnCheckbox = '#pdf_delivery_on';
+    this.shippedOnCheckbox = '#shipped_on';
+    this.paidOnCheckbox = '#paid_on';
+    this.deliveryOnCheckbox = '#delivery_on';
+    this.saveButton = '#order_state_form_submit_btn';
```
Comment (SimonGrn) [inline tests/UI/pages/BO/shopParameters/orderSettings/statuses/add.js:24]: This is beautiful

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/orders/view.js (hunk 1 above)
Comment (matks) [reply tests/UI/pages/BO/orders/view.js]:
```suggestion
  async doesStatusExist(page, statusName) {
```
Indeed 😉 `is` must be followed by a noun like `isName` `isInteger` `isValid` ; but the verb here is `Exist` so `does` must be used

File: tests/UI/pages/BO/shopParameters/orderSettings/index.js (hunk 2 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/shopParameters/orderSettings/index.js:32]: Yes

File: tests/UI/pages/BO/orders/view.js (hunk 1 above)
Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/orders/view.js]: Done thnx :+1:

**Review by matthieu-rolland — APPROVED** [review]

**Review by SimonGrn — APPROVED** [review]

**Pull request merged**